### PR TITLE
added command to get the active folder of a workspace (#1715)

### PR DESCRIPTION
**Pull request by guestieng**
## This change addresses item #1715

The following changes are proposed:

- adding command `cmake.activeFolder` for allowing to get the active folder of a workspace

## The purpose of this change

- This change can be helpful when working with tasks/macros/multi-commands.

### CODE REVIEW

**Comment by andreeis** [comment]
Thank you for your contribution. At a quick glance it looks good. We intend to test and merge it in the repository after we release CMake Tools 1.7 next week.

File: src/extension.ts [hunk 1]
```diff
@@ -1137,6 +1137,14 @@ class ExtensionManager implements vscode.Disposable {
     await logging.showLogFile();
   }
 
+  activeFolder() : string  {
+    let thisActiveFolder: string = '';
+    if(this._folders.activeFolder){
+      thisActiveFolder = this._folders.activeFolder.folder.name;
+    }
+    return thisActiveFolder;
+  }
```
Comment (xisui-MSFT) [inline src/extension.ts:1146]: You can do this which is equivalent:
```
activeFolder() : string { return this._folders.activeFolder?.folder.name || ''; }
```

Also, do you only need the name or the full path? Will there be problems when there are multiple folders with the same name?

File: package.json [hunk 2]
```diff
@@ -79,10 +80,16 @@
...
     "workspaceContains:.vscode/cmake-kits.json"
-    ],
+  ],
   "main": "./dist/main",
   "contributes": {
     "commands": [
+      {
+        "command": "cmake.activeFolder",
+        "title": "Get active folder",
+        "when": "cmake:enableFullFeatureSet",
```
Comment (xisui-MSFT) [inline package.json:90]: You need to add 'when' to the menus/commandPalette list to make it effective (actually it does nothing in contributes/commands).

That being said, I'm not sure if enableFullFeatureSet is required here.

**Review by xisui-MSFT — DISMISSED** [review]

File: src/extension.ts (hunk 1 above)
Comment (guestieng) [reply src/extension.ts]: This is a point. For now both are implemented, the active folders name and path, respectively.

File: src/extension.ts [hunk 3]
```diff
@@ -1506,7 +1513,9 @@ async function setup(context: vscode.ExtensionContext, progress?: ProgressHandle
 
   // List of functions that will be bound commands
   const funs: (keyof ExtensionManager)[] = [
-    "useCMakePresets",
+    'activeFolderName',
+    'activeFolderPath',
+	  "useCMakePresets",
```
Comment (xisui-MSFT) [inline src/extension.ts:1518]: Indentation?

Comment (guestieng) [reply src/extension.ts]: Interesting, in Vscode (1.54.1) this is not visible.

**Comment by guestieng** [comment]
All requested changes are actually implemented. Still, the status display says "Changes requested" by now. 
This seems a bit misleading. 
Therefore, I kindly request a final review.

**Comment by xisui-MSFT** [comment]
Sorry for the confusion. I have reset the review. Formal approval would be given once 1.8.0 release starts to accept PRs.

**Comment by bobbrow** [comment]
I'll merge this in and fix the remaining issues with linting/documentation.

**Pull request merged**